### DISCUSSION TITLE
Enabling a larger max_part_size for multipart forms

### DISCUSSION THREAD

**Discussion by taiyosogawa** [encode/starlette#2765] [Q&A]
I'm trying to figure out how to increase the max_part_size for multipart forms, which appears to be fixed to 1MB since [this commit](https://github.com/encode/starlette/commit/fd038f3070c302bff17ef7d173dbb0b007617733#diff-59eaf7e36ef8b186735f99bca714455f02f04cee26e014dc89120417996d35b2R119).

@Kludex, thanks for putting that security feature in - I'm wondering if there's a good way to configure it as-is or what your thoughts would be to add a way to configure it for specific scenarios?

My use case is to integrate with Twilio SendGrid's inbound parse webhooks. Whenever an email is received, SendGrid will POST the contents and all the attachments of that email to my FastAPI server. The attachments are expected to exceed 1MB and they are sent as a multi-part form ([docs with example payload](https://www.twilio.com/docs/sendgrid/for-developers/parsing-email/setting-up-the-inbound-parse-webhook#example-raw-payload)). I don't have any control over how they push that data, otherwise I might find an alternative method.

Thank you in advance for your thoughts and suggestions!

**Comment by Kludex** [comment]
I think we need to move this logic to the `python-multipart`. But... I'm not sure if we should have separate limits for file and field. Maybe just `max_part_size`?

**Comment by Nixxs** [comment]
I've also just run into this issue, the docs say you can set the limit using 

```python
await request.form(
    max_part_size=1000000
)
```

but that consumes the request and you can't get it later on so you have to call this first

`await request.body()`

which caches it? but still the request.form( max_part_size=11000) doesnt seem to work anyway.
